python/google/protobuf/pyext/descriptor.cc

```diff
@@ -100,18 +137,18 @@ bool _CalledFromGeneratedFile(int stacklevel) {
     return false;
   }
   while (stacklevel-- > 0) {
-    frame = frame->f_back;
+    frame = PyFrame_GetBack(frame);
     if (frame == nullptr) {
       return false;
     }
   }
 
-  if (frame->f_code->co_filename == nullptr) {
+  if (PyFrame_GetCode(frame)->co_filename == nullptr) {
```

Comment:
every use of `frame->f_code` is used to access `co_filename`.  Perhaps we should make the helper do that instead.

Reply:
The helper added above is only to provide backwards compatibility. The `PyFrame_GetCode` that was added is meant to match the same behavior as the new function in Python 3.11